### ISSUE TITLE
Switch HDBSCAN import from standalone package to scikit-learn

### ISSUE THREAD

**Issue by MinhHaDuong** [MinhHaDuong/Oeconomia-Climate-finance#608]
## Problem

`scripts/clustering_methods.py:53` imports the standalone `hdbscan` package, which is unmaintained (absorbed into scikit-learn since v1.3). The package is not installed in the current environment, causing 4 test failures in `test_compare_clustering.py`.

## Fix

In `cluster_hdbscan()` (line 51–61), replace:

```python
import hdbscan
clusterer = hdbscan.HDBSCAN(
    min_cluster_size=min_cluster_size,
    min_samples=min_samples,
    metric="euclidean",
    core_dist_n_jobs=1,
)
```

with:

```python
from sklearn.cluster import HDBSCAN
clusterer = HDBSCAN(
    min_cluster_size=min_cluster_size,
    min_samples=min_samples,
    metric="euclidean",
    n_jobs=1,
)
```

Two changes: import path, and `core_dist_n_jobs` → `n_jobs`. The rest of the API (`min_cluster_size`, `min_samples`, `metric`, `fit_predict()`) is identical.

If `hdbscan` is listed in `pyproject.toml` dependencies, remove it.

## First test

Run `uv run pytest tests/test_compare_clustering.py -v` — all 4 currently failing HDBSCAN tests should pass.

## Scope

- `scripts/clustering_methods.py` — the only file that imports `hdbscan`
- `pyproject.toml` — remove standalone `hdbscan` dependency if present
